x-pack/plugins/observability/server/services/slo/find_slo_groups.ts

```diff
@@ -161,14 +231,38 @@ export class FindSLOGroups {
               degrading: bucket.degrading?.doc_count,
               noData: bucket.noData?.doc_count,
             },
-          },
-        ];
-      }, [] as Array<Record<'group' | 'groupBy' | 'summary', any>>) ?? [];
-    return findSLOGroupsResponseSchema.encode({
-      page: pagination.page,
-      perPage: pagination.perPage,
+          };
+        })
+        .filter((item) => item && item.summary.total > 0) as Array<
+        Record<'group' | 'groupBy' | 'summary', any>
+      >) ?? [];
+    return {
+      page: 1,
+      perPage: total,
       total,
       results,
+    };
+  }
+  async findSavedQueries() {
+    const savedQuery = await this.context.savedQuery;
+
+    // @ts-expect-error
+    const { savedQueries } = await savedQuery.getAll();
```

Comment:
I don't have data on this, but I do wonder what the performance of this could be for clusters that have a VERY large number of saved queries present. Is there some way we could scope it down before issuing the request to ES?